modules/packages/Image.chpl

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2024-2024 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+The Image module provides a way to write arrays of pixels to an output image format.
+
+For example, the following code creates a 3x3 array of pixels and writes it to
+a BMP file. The array is then scaled by a factor of 2 (creating a 9x9 image)
+and written to a second BMP file.
+
+.. code-block:: chapel
+
+   use Image;
+
+   var arr: [1..3, 1..3] int;
+
+   arr[1, ..] = [0xFF0000, 0x00FF00, 0x0000FF];
+   arr[2, ..] = [0x00FF00, 0x0000FF, 0xFF0000];
+   arr[3, ..] = [0x0000FF, 0xFF0000, 0x00FF00];
+
+   writeImage("pixels.bmp", imageType.bmp, arr);
+   writeImage("pixels2.bmp", imageType.bmp, scale(arr, 2));
+
+
+*/
+@unstable("Image is unstable")
+module Image {
+
+  private use IO, Math;
+
+
+  //
+  // Configuration params/types
+  // (Override defaults on compiler line using -s<cfg>=<val>)
+  //
+  @chpldoc.nodoc
+  config type pixelType = int;
+  @chpldoc.nodoc
+  config param bitsPerColor = 8;
+
+  //
+  // Define config-dependent types and params.
+  //
+  @chpldoc.nodoc
+  param colorMask = (0x1 << bitsPerColor) - 1;
+  @chpldoc.nodoc
+  type colorType = uint(bitsPerColor);
+
+
+  //
+  // set helper params for colors
+  //
+  @chpldoc.nodoc
+  param red = 0,        // names for referring to colors
```

Comment:
This is the perfect use case for an enum followed by a use, including the auto-numbered 'numColors'